ios/RNFIRMessaging.m

```diff
@@ -247,39 +286,31 @@ - (void) onTokenRefresh
   [_bridge.eventDispatcher sendDeviceEventWithName:@"FCMTokenRefreshed" body:[[FIRInstanceID instanceID] token]];
 }
 
-RCT_EXPORT_METHOD(requestPermissions)
-{
+RCT_REMAP_METHOD(requestPermissions,
+                 resolver:(RCTPromiseResolveBlock)resolve
+                 rejecter:(RCTPromiseRejectBlock)reject){
   if (RCTRunningInAppExtension()) {
+    reject(RNFIRErrorUnableToRequestPermissions, nil, RCTErrorWithMessage(@"Requesting push notifications is currently unavailable in an app extension"));
+  }
+  if (_requestPermissionsResolveBlock != nil) {
+    RCTLogError(@"Cannot call requestPermissions twice before the first has returned.");
     return;
   }
-  if (floor(NSFoundationVersionNumber) <= NSFoundationVersionNumber_iOS_9_x_Max) {
-    UIUserNotificationType allNotificationTypes =
-    (UIUserNotificationTypeSound | UIUserNotificationTypeAlert | UIUserNotificationTypeBadge);
-    UIApplication *app = RCTSharedApplication();
-    if ([app respondsToSelector:@selector(registerUserNotificationSettings:)]) {
-      //iOS 8 or later
-      UIUserNotificationSettings *notificationSettings =
-      [UIUserNotificationSettings settingsForTypes:(NSUInteger)allNotificationTypes categories:nil];
-      [app registerUserNotificationSettings:notificationSettings];
-    } else {
-      //iOS 7 or below
-      [app registerForRemoteNotificationTypes:(NSUInteger)allNotificationTypes];
-    }
+  [self startObservingRegistrationNotifications];
+  _requestPermissionsResolveBlock = resolve;
+  UIUserNotificationType allNotificationTypes =
+  (UIUserNotificationTypeSound | UIUserNotificationTypeAlert | UIUserNotificationTypeBadge);
+  UIApplication *app = RCTSharedApplication();
+  if ([app respondsToSelector:@selector(registerUserNotificationSettings:)]) {
+    //iOS 8 or later
+    UIUserNotificationSettings *notificationSettings =
+    [UIUserNotificationSettings settingsForTypes:(NSUInteger)allNotificationTypes categories:nil];
+    [app registerUserNotificationSettings:notificationSettings];
   } else {
-    // iOS 10 or later
-#if defined(__IPHONE_10_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_10_0
```

Comment:
looks like this piece of code got removed. can you bring this back?
also you can skip `[self startObservingRegistrationNotifications];` stuff because it has completionHandler itself